### PR TITLE
feat(codegen-scala): generate empty main

### PR DESCRIPTION
**Pull request by raboof**

### CODE REVIEW

File: codegen/scala-gen/src/main/scala/com/akkaserverless/codegen/scalasdk/SourceGenerator.scala [hunk 1]
```diff
@@ -19,24 +19,80 @@ package com.akkaserverless.codegen.scalasdk
...
 
+  val managedComment = """/* This code is managed by Akka Serverless tooling.
+                         | * It will be re-generated to reflect any changes to your protobuf definitions.
+                         | * DO NOT EDIT
+                         | */""".stripMargin
+
+  val unmanagedComment = """|/* This code was generated by Akka Serverless tooling.
+                            | * As long as this file exists it will not be re-generated.
+                            | * You are free to make changes to this file.
+                            | */""".stripMargin
```
Comment (patriknw) [inline codegen/scala-gen/src/main/scala/com/akkaserverless/codegen/scalasdk/SourceGenerator.scala:34]: this has moved in https://github.com/lightbend/akkaserverless-java-sdk/pull/422

File: codegen/scala-gen/src/main/scala/com/akkaserverless/codegen/scalasdk/SourceGenerator.scala [hunk 2]
```diff
@@ -19,24 +19,80 @@ package com.akkaserverless.codegen.scalasdk
...
+        _.replaceFirst("\\.[^.]*$", "")
+          .split("\\.")
+          .toList)
+      .toSet
+    if (packages.isEmpty) throw new IllegalArgumentException("Nothing to generate!")
+    longestCommonPrefix(packages.head, packages.tail)
+  }
+
+  @tailrec
+  def longestCommonPrefix(
```
Comment (patriknw) [inline codegen/scala-gen/src/main/scala/com/akkaserverless/codegen/scalasdk/SourceGenerator.scala:83]: this has moved to SourceGeneratorUtils in https://github.com/lightbend/akkaserverless-java-sdk/pull/422

File: codegen/scala-gen/src/main/scala/com/akkaserverless/codegen/scalasdk/SourceGenerator.scala [hunk 3]
```diff
@@ -19,24 +19,80 @@ package com.akkaserverless.codegen.scalasdk
...
+      s"""
+         |package ${mainPackage.mkString(".")}
+         |
+         |object Main extends App {
+         |  println("Hello, world!")
+         |}
+         |""".stripMargin)
+  }
+
+  def mainPackageName(classNames: Iterable[String]): List[String] = {
```
Comment (patriknw) [inline codegen/scala-gen/src/main/scala/com/akkaserverless/codegen/scalasdk/SourceGenerator.scala:71]: this has moved to SourceGeneratorUtils in #422

**Review by patriknw — COMMENTED** [review]
looking good, but would be best to merge #422 first

**Review by octonato — APPROVED** [review]

**Review by patriknw — APPROVED** [review]
LGTM

**Pull request merged**